client/app/services/data-source.js

```diff
@@ -3,7 +3,7 @@ function DataSource($resource) {
     get: { method: 'GET', cache: false, isArray: false },
     query: { method: 'GET', cache: false, isArray: true },
     test: { method: 'POST', cache: false, isArray: false, url: 'api/data_sources/:id/test' },
-    getSchema: { method: 'GET', cache: true, isArray: true, url: 'api/data_sources/:id/schema' },
+    getSchema: { method: 'GET', cache: false, isArray: true, url: 'api/data_sources/:id/schema' },
```

Comment:
Yes, we can add separate `refreshSchema` method which will do two things:
1) Perform non cached request with `refresh=true` in url.
2) Remove cache for `getSchema`

Like here: http://stackoverflow.com/a/25124772

The main drawback here that we'll get extra request with next `getSchema` call.

Unfortunately, we don't have much control on how Angular stores values in cache (it uses full url as cache key: https://github.com/angular/angular.js/blob/v1.5.11/src/ng/http.js#L1251). It seems like there is no simple way to perform request on one url (with `refresh=true`), but renew cache data for another url.